src/App.jsx

```diff
@@ -1,14 +1,65 @@
 import './App.css';
+import ChatLog from './components/ChatLog';
+import ColorChoice from './components/ColorChoice';
+import DATA from './data/messages.json';
+import { useState } from 'react';
 
 const App = () => {
+  const [entryData, setEntryData] = useState(DATA);
+  const [localColor, setLocalColor] = useState('green');
+  const [remoteColor, setRemoteColor] = useState('blue');
+
+  const LOCAL_SENDER = entryData[0].sender;
+  const REMOTE_SENDER = entryData[1].sender;
+
+  const handleColorChange = (sender, color) => {
+    if (sender === LOCAL_SENDER){
+      setLocalColor(color);
+    } else if (sender === REMOTE_SENDER) {
+      setRemoteColor(color);
+    }
+  };
+
+  const updateEntryLikedState = (entryId) => {
+    setEntryData(entries => {
+      return entries.map(entry => {
```

Comment:
Nice use of map here to both handle making a new list so that React sees the message data has changed, and make new data for the clicked message with its like status toggled.